prov/efa/src/efa_domain.c

```diff
@@ -74,6 +74,12 @@ static int efa_domain_close(fid_t fid)
 	if (ret)
 		return ret;
 
+	if (rxr_env.enable_shm_transfer) {
```

Comment:
We are opening shm domain in rxr, while releasing shm domain in efa, which looks a little bit weird. 
Can we release shm domain before efa domain in rxr_domain_close()?

Reply:
I tried that, but it did not work because shm_domain is bind to some resource in efa_domain. Those resources have be closed before we can close shm_domain. I will look into it further.

Reply:
I tried again, and can confirm that shm_domain was linked to some resource in efa_domain, therefore can only be closed after other resources was closed, but has to be closed before ```free(domain)```, so it has to be inside ```efa_domain_open()```

Reply:
> shm_domain was linked to some resource in efa_domain

Hmm. do you know what these resources are?

Reply:
It is the mr_cache somehow connected to shm_domain. Inside ```efa_domain_close``` there is a call to ```ofi_mr_cache_cleanup```.  I tried close ```shm_domain``` before that call, and close will fail with ```resource busy```.

Reply:
Talked to @wzamazon offline. Since we will merge rxr and efa domain eventually, we can leave the fix as it is now. I will open another PR later if I can quickly find another decent solution.